x-pack/filebeat/input/azureeventhub/sanitizers.go

```diff
@@ -0,0 +1,201 @@
+		// legacy sanitizer don't need to be initialized
+		switch sanitizationOption(opt) {
+		case newLines:
+			sanitizers = append(sanitizers, &newLinesSanitizer{})
+		case singleQuotes:
+			sanitizers = append(sanitizers, &singleQuotesSanitizer{})
+		}
+	}
+
+	return sanitizers, nil
+}
+
+// ----------------------------------------------------------------------------
+// New line sanitizer
+// ----------------------------------------------------------------------------
+
+// newLinesSanitizer replaces new lines with spaces.
+//
+// This sanitizer is used to remove new lines inside JSON strings.
+type newLinesSanitizer struct{}
+
+func (s *newLinesSanitizer) Sanitize(jsonByte []byte) []byte {
+	return bytes.ReplaceAll(jsonByte, []byte("\n"), []byte{})
+}
+
+// Init is a no-op for the newLinesSanitizer.
+func (s *newLinesSanitizer) Init() error {
+	return nil
+}
+
+// ----------------------------------------------------------------------------
+// Single quote sanitizer
+// ----------------------------------------------------------------------------
+
+// singleQuotesSanitizer replaces single quotes with double quotes.
+type singleQuotesSanitizer struct{}
+
+func (s *singleQuotesSanitizer) Sanitize(jsonByte []byte) []byte {
+	var result bytes.Buffer
+	var prevChar byte
+
+	inDoubleQuotes := false
+
+	for _, r := range jsonByte {
+		if r == '"' && prevChar != '\\' {
+			inDoubleQuotes = !inDoubleQuotes
+		}
+
+		if r == '\'' && !inDoubleQuotes {
+			result.WriteRune('"')
+		} else {
+			result.WriteByte(r)
+		}
+		prevChar = r
+	}
+
+	return result.Bytes()
+}
+
+// Init is a no-op for the singleQuotesSanitizer.
+func (s *singleQuotesSanitizer) Init() error {
+	return nil
+}
+
+// ----------------------------------------------------------------------------
+// Replace all sanitizer
+// ----------------------------------------------------------------------------
+
+// replaceAllSanitizer replaces all occurrences of a regex pattern with a replacement string.
+type replaceAllSanitizer struct {
+	re          *regexp.Regexp
+	replacement string
+	spec        map[string]interface{}
+}
+
+func (s *replaceAllSanitizer) Sanitize(jsonByte []byte) []byte {
+	if s.re == nil {
+		return jsonByte
+	}
+
+	return s.re.ReplaceAll(jsonByte, []byte(s.replacement))
+}
+
+func (s *replaceAllSanitizer) Init() error {
+	if s.spec == nil {
+		return errors.New("missing sanitizer spec")
+	}
+
+	if _, ok := s.spec["pattern"]; !ok {
+		return errors.New("missing regex pattern")
+	}
+
+	if _, ok := s.spec["pattern"].(string); !ok {
+		return errors.New("regex pattern must be a string")
+	}
+
+	re, err := regexp.Compile(s.spec["pattern"].(string))
+	if err != nil {
+		return err
+	}
+
+	s.re = re
+
+	if _, ok := s.spec["replacement"]; !ok {
+		return errors.New("missing replacement format")
+	}
+
+	if _, ok := s.spec["replacement"].(string); !ok {
+		return errors.New("replacement format must be a string")
+	}
+
+	s.replacement = s.spec["replacement"].(string)
+
+	return nil
+}
```

Comment:
Do you think its worth creating a helper function that gets a key from the `spec` map?

```suggestion
func (s *replaceAllSanitizer) Init() error {
    if s.spec == nil {
        return errors.New("missing sanitizer spec")
    }

    pattern, err := getStringFromSpec(s.spec, "pattern")
    if err != nil {
        return fmt.Errorf("invalid pattern: %w", err)
    }

    re, err := regexp.Compile(pattern)
    if err != nil {
        return fmt.Errorf("invalid regex pattern: %w", err)
    }
    s.re = re

    replacement, err := getStringFromSpec(s.spec, "replacement")
    if err != nil {
        return fmt.Errorf("invalid replacement: %w", err)
    }
    s.replacement = replacement

    return nil
}

func getStringFromSpec(spec map[string]interface{}, key string) (string, error) {
    value, ok := spec[key]
    if !ok {
        return "", fmt.Errorf("missing %s", key)
    }

    strValue, ok := value.(string)
    if !ok {
        return "", fmt.Errorf("%s must be a string", key)
    }

    return strValue, nil
}
```